src/explore_tools.py

```diff
@@ -0,0 +1,58 @@
+
+from os import listdir
+from os.path import join, isfile, isdir
+
+import config
+from models.config import Config
+
+def glob_all(namespace):
+    def __glob_helper(extra_args, config_exists, dir_path, parent_value):
+        collector = parent_value
+        if config_exists:
+            collector += [dir_path]
+        return collector
+
+    collector = []
+    each_dir(namespace,
+             __glob_helper,
+             parent_value = collector)
+    collector = list(map(join_config, collector))
+    return collector
+
+def single_level(namespace):
```

Comment:
`single_level` isn't very descriptive as well. Possible name it `inspect_namespace`

Reply:
Also done here https://github.com/alces-software/adminware/pull/99/commits/d8d2955436d53392d83c52ce181865bb36cf5878

Reply:
I don't think `single` is required here. It is implied to be singular as the name is `inspect_namespace` instead of `inspect_namespaces`

Reply:
Do you feel like the singular-ness is enough to demonstrate that the function doesn't explore `namespace`'s sub-namespace(s)? happy to change it if you think so